### PR TITLE
Add button_dialog, including docs/examples

### PR DESCRIPTION
**Pull request by sreyemnayr**
Provides the ability to render several buttons with values of your choosing.
Updated docs and examples to show how it works.
button

### CODE REVIEW

**Comment by sreyemnayr** [comment]
Sorry, but I don't have any idea how to write tests for this... But it works!

File: docs/pages/dialogs.rst [hunk 1]
```diff
@@ -64,6 +64,30 @@ selection.
...
+tuples, providing the label (first) and return value if clicked (second).
+
+.. code:: python
+
+    from prompt_toolkit.shortcuts.dialogs import button_dialog
+
+    result = button_dialog(
+        title='Button dialog example',
+        text='Do you want to confirm?',
+        buttons=(
```
Comment (jonathanslenders) [inline docs/pages/dialogs.rst]: Could you make this a list of tuples? 
See also this page for more about the difference: https://nedbatchelder.com/blog/201608/lists_vs_tuples.html

Also, I like spaces after the commas in the tuples. ;)

File: examples/dialogs/button_dialog.py [hunk 2]
```diff
@@ -0,0 +1,24 @@
+#!/usr/bin/env python
+"""
+Example of confirmation (yes/no) dialog window.
```
Comment (jonathanslenders) [inline examples/dialogs/button_dialog.py]: Could you fix this docstring?

**Comment by sreyemnayr** [comment]
Done!  Truthfully, the yes_no_dialog could just call the button_dialog now to maintain DRY...  
Glad to have been able to contribute (and to learn about functools.partial today) :)

**Comment by jonathanslenders** [comment]
Thanks a lot for this! It's very much appreciated. Feel free to create another merge request to rewrite ``yes_no_dialog`` building on top of this.

It also reminds me that I should fix the background of the buttons when they are selected. I removed the 'focussed' class from the style in the ``Window`` because that was not the right place to add this, but it should be added somewhere else.

Cheers!

**Comment by sreyemnayr** [comment]
I was actually going to ask you how to do this!

________________________________

**Comment by jonathanslenders** [comment]
See here: https://github.com/jonathanslenders/python-prompt-toolkit/commit/3c8a29ed52203b0f470bc24c01fd735c60cd029d

**Comment by sreyemnayr** [comment]
Nice! I’ll knock that yes_no_dialog piece out in a bit. Such a cool project, I’m honored to contribute!

Ryan Meyers
________________________________

**Pull request merged**